test/benchmarks/include/vikunja/bench/memory.hpp

```diff
@@ -0,0 +1,213 @@
+/* Copyright 2022 Simeon Ehrig
+ *
+ * This file is part of vikunja.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/alpaka.hpp>
+
+#include <algorithm>
+#include <vector>
+
+namespace vikunja::bench
+{
+    template<typename TData>
+    class IotaFunctor
+    {
+    private:
+        TData const m_begin;
+        TData const m_increment;
+
+    public:
+        //! Functor for iota implementation with generic data type.
+        //!
+        //! \tparam TData Type of each element
+        //! \param begin Value of the first element.
+        //! \param increment Distance between two elements.
+        IotaFunctor(TData const begin, TData const increment) : m_begin(begin), m_increment(increment)
```

Comment:
`begin` is a bit confusing (because of `std::begin`). Maybe call it `init` or `value` like in `std::iota`'s interface?